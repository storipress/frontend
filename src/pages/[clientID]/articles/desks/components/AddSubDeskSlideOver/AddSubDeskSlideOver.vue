<script lang="ts" setup>
import type { PropType } from 'vue'
import { Buttons, Inputs } from '@storipress/core-component'
import { useForm } from 'vee-validate'
import type { DeskInterface, EventSubmitDataInterface } from './definition'
import SlideOver from '~/components/SlideOver'
import { createDeskNameSchema } from '~/composables'

const props = defineProps({
  show: {
    type: Boolean,
    required: true,
    default: false,
  },
  loading: {
    type: Boolean,
    required: true,
    default: false,
  },
  parent: {
    type: Object as PropType<DeskInterface>,
    required: true,
  },
})
const emit = defineEmits<{
  (event: 'close'): void
  (event: 'submit', newSubDesk: EventSubmitDataInterface): void
}>()

const name = ref('')
const description = ref('')
whenever(
  () => !props.show,
  () => {
    name.value = ''
    description.value = ''
  },
)

const { handleSubmit } = useForm({
  validationSchema: createDeskNameSchema('name'),
})
const submit = handleSubmit(() => {
  emit('submit', {
    name: name.value,
    description: description.value,
    parentId: props.parent.id,
  })
})
</script>

<template>
  <SlideOver
    class="add-sub-desk-slide-over"
    :show="show"
    :title="`Create New Sub-Desk in ${parent?.name}`"
    :click-outside-to-close="!loading"
    @close="$emit('close')"
    @submit="!loading && submit()"
  >
    <SectionContent
      sub-title="Subdesk settings"
      content="The subdesk name is used in your live site for section headings and navigation."
    >
      <template #content>
        <div>
          <Inputs v-model="name" label="Subdesk name" html-type="text" html-name="name" />
        </div>
        <div class="mt-4">
          <Inputs v-model="description" label="Subdesk description" html-type="text" html-name="description" />
        </div>
      </template>
    </SectionContent>
    <template #footer>
      <Buttons type="main" color="primary" html-type="submit" :is-loading="loading"> Create new desk </Buttons>
    </template>
  </SlideOver>
</template>

<style lang="scss">
.add-sub-desk-slide-over {
  .slide-over__footer {
    @apply justify-end;
  }
}
</style>
