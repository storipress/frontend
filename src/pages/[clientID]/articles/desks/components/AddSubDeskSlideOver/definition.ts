export interface DeskInterface {
  id: string
  name: string
}

export interface EventSubmitDataInterface {
  name: string
  description: string
  parentId: string
}
