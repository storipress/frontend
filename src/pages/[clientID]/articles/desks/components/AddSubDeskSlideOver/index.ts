export { default } from './AddSubDeskSlideOver.vue'
export * from './definition'
