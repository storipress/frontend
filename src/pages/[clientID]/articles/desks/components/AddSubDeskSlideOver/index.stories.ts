import type { Story } from '@storybook/vue3'
import { reactive } from 'vue'
import { userEvent, within } from '@storybook/test'
import { action } from '@storybook/addon-actions'
import type { EventSubmitDataInterface } from './index'
import AddSubDeskSlideOver from './index'

export default {
  title: 'Pages/Articles/Components/desk settings',
  decorators: [
    (story) => ({
      components: { story },
      template: '<div style="width:100vw;height:100vh;margin:-16px;"><story /></div>',
    }),
  ],
}

const Template: Story = (args) => ({
  components: { AddSubDeskSlideOver },
  template: `
    <div>
      <AddSubDeskSlideOver
        v-bind="data"
        @close="closeSlideOver"
        @submit="submit"
      />
      <button data-testid="trigger" @click="data.show = !data.show" class="bg-gray-300">
        Click me
      </button>
      <pre>
        name: {{data.name}}
      </pre>
    </div>
  `,
  setup() {
    const data = reactive(args)
    return {
      data,
      closeSlideOver() {
        action('close')()
        data.show = false
      },
      submit(eventData: EventSubmitDataInterface) {
        action('submit')(eventData)
        data.name = eventData.name
        data.loading = true
        setTimeout(() => (data.loading = false), 1000)
      },
    }
  },
})

export const CreateSubDesk = Template.bind({})
CreateSubDesk.args = {
  show: false,
  loading: false,
  parent: { id: '0', name: 'Business' },
}
CreateSubDesk.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  const trigger = canvas.getByTestId('trigger')
  await userEvent.click(trigger)
}
