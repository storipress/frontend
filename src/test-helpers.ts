import { NotificationsPlugin } from '@storipress/core-component'
import { createTestingPinia } from '@pinia/testing'
import { ApolloClients, provideApolloClients } from '@vue/apollo-composable'
import type { RenderOptions, RenderResult } from '@testing-library/vue'
import generatedRoutes from 'virtual:generated-pages'
import { render as vueRender } from '@testing-library/vue'
import type { RouteRecordRaw, Router } from 'vue-router'
import { createRouter, createWebHashHistory } from 'vue-router'
import { defu } from 'defu'
import { defineComponent, h } from 'vue'
import { createHead } from '@unhead/vue'
import { plugin as Slicksort } from 'vue-slicksort'
import type { TypedDocumentNode } from '@apollo/client/core'
import type { PartialDeep } from 'type-fest'
import tailwindcss from 'tailwindcss'
import postcss from 'postcss'
import type { GraphQLError } from 'graphql'
import { setActivePinia } from 'pinia'
import { server } from '../test/server'
import { KEY } from '../test/setupFiles/constant'
import type { MockHandler } from './mocks/define-graphql-mock'
import { defineGraphQLMock } from './mocks/define-graphql-mock'
import { key as confirmModelKey } from '~/components/ConfirmModalProvider/definition'
import { enableFeatures as _enableFeatures } from '~/lib/feature-flag'
import { clients } from '~/lib/apollo'

interface CustomRenderOptions extends RenderOptions {
  routes?: boolean | RouteRecordRaw[]
  stubActions?: boolean
}

interface CustomRenderResult extends RenderResult {
  router: ReturnType<typeof createRouter>
}

export const mockAxiomIngest = vi.fn()

export function render(
  // skipcq: JS-0323
  component: any,
  options: CustomRenderOptions & { css: true },
): Promise<CustomRenderResult> & CustomRenderResult
// skipcq: JS-0323
export function render(component: any, options?: CustomRenderOptions & { css?: false }): CustomRenderResult
// skipcq: JS-0323
export function render(component: any, options?: CustomRenderOptions & { css?: boolean }) {
  const routes = options?.routes
    ? options.routes === true
      ? generatedRoutes
      : options.routes
    : [
        { path: '/', component: defineComponent({ setup: () => () => h('div', 'Home') }) },
        { path: '/**', component: defineComponent({ setup: () => () => h('div', 'CatchAll') }) },
      ]
  const router = createRouter({
    history: createWebHashHistory(),
    routes,
  })

  const renderOptions: RenderOptions = {
    ...options,
    global: {
      ...options?.global,
      plugins: [
        ...(options?.global?.plugins ?? []),
        NotificationsPlugin,
        Slicksort,
        setupTestPinia(options?.stubActions),
        createHead(),
        router,
        (app) => {
          app.provide(KEY, 'secret-key')
          app.provide(ApolloClients, clients)
          app.provide(confirmModelKey, { confirm: vi.fn(), modalList: [] })
        },
      ],
    },
  }
  const res = vueRender(component, renderOptions)

  // when css is true, create a new instance of tailwindcss and postcss
  if (options?.css) {
    const lazyResult = postcss([
      tailwindcss({
        content: [{ raw: res.container.innerHTML }],
      }),
    ]).process('@tailwind utilities;', { from: undefined })
    const promise: Promise<RenderResult> & RenderResult = lazyResult.then((result) => {
      const style = document.createElement('style')
      style.innerHTML = result.css
      res.container.append(style)
      return {
        ...res,
        router,
      }
    }) as Promise<CustomRenderResult> & CustomRenderResult

    return new Proxy(promise, {
      get(target, key) {
        if (key === 'then') {
          const then = target.then.bind(target)
          return (...args: Parameters<typeof then>) => {
            then(...args)
          }
        }
        throw new Error('Please await the returned promise if you are using `css: true`')
      },
    })
  }

  return {
    ...res,
    router,
  }
}

interface Composable<Params extends unknown[], Return> {
  (...params: Params): Return
}

export function runComposable<Params extends unknown[], Return>(
  composable: Composable<Params, Return>,
  ...params: Params
): { result: Return; router: Router } {
  const router = createRouter({
    history: createWebHashHistory(),
    routes: [{ path: '/', component: defineComponent({ setup: () => () => h('div', 'Home') }) }],
  })

  const app = createApp({
    setup: () => () => h('div'),
  })

  app.use(NotificationsPlugin)
  app.use(Slicksort)
  app.use(setupTestPinia())
  app.use(createHead())
  app.use(router)
  app.provide(KEY, 'secret-key')
  app.provide(ApolloClients, clients)
  app.provide(confirmModelKey, { confirm: vi.fn(), modalList: [] })

  return {
    result: app.runWithContext(() => composable(...params)),
    router,
  }
}

export function setupApolloClient() {
  provideApolloClients(
    new Proxy(
      {},
      {
        get: () => {
          return clients.default
        },
      },
    ),
  )
  resetApolloCache()
}

export function resetApolloCache() {
  clients.default.cache.reset({})
}

export function setupTestPinia(stubActions?: boolean) {
  return createTestingPinia({
    stubActions,
    initialState: {
      client: {
        clientID: 'D6RX98VXN',
        defaultClientID: 'default',
        token: 'cAA6LoWllmqdXaYYid1AxpEDy5liOBGp5ppKPNKM',
        isResetPassword: false,
      },
      'search-condition': {
        text: undefined,
        people: undefined,
        tags: undefined,
        range: undefined,
      },
      site: {
        site: {
          name: "Joe's Blog",
          timezone: 'Asia/Taipei',
          email: 'test@storipress.com',
          socials: '{"Facebook":"www.facebook.com\\/Storipress"}',
        },
        siteTutorials: {
          setSocialConnect: true,
          setPublicationDetail: true,
          setCreateDesks: true,
          setDomain: true,
        },
      },
      me: {
        me: {
          id: 151,
          role: 'owner',
        },
      },
      integration: {
        integrations: [
          { activated_at: null, data: '{"shortname":null}', key: 'disqus', order: 6 },
          {
            activated_at: null,
            data: '[]',
            key: 'webflow',
            order: 7,
          },
        ],
      },
      knownErrors: {
        errors: [],
      },
    },
  })
}

export function setupTestEnv() {
  setActivePinia(setupTestPinia())
  setupApolloClient()
}

export function useTestEnv() {
  beforeEach(setupTestEnv)
}

export function useFakeTimer(now?: number | Date) {
  beforeEach(() => {
    vi.useFakeTimers({ now })
  })

  afterEach(() => {
    vi.useRealTimers()
  })
}

export function mockResponse<QueryOutput extends Record<string, any>>(
  doc: TypedDocumentNode<QueryOutput, any>,
  data: QueryOutput,
) {
  const handler = defineGraphQLMock(doc, data, (output, res, ctx) => res.once(ctx.data(output)))

  server.use(handler)
}

export function mockResponseOnce<QueryOutput extends Record<string, any>, Variables extends Record<string, any>>(
  originalHandler: MockHandler<QueryOutput, Variables>,
  patchedQuery: PartialDeep<QueryOutput> = {} as PartialDeep<QueryOutput>,
) {
  const handler = defineGraphQLMock<QueryOutput, Variables>(
    originalHandler.doc,
    (req) => {
      return defu(patchedQuery, originalHandler.resolveOutput(req)) as QueryOutput
    },
    (output, res, ctx) => res.once(ctx.data(output)),
  )

  server.use(handler)
}

export function mockResponseError<QueryOutput extends Record<string, any>, Variables extends Record<string, any>>(
  originalHandler: MockHandler<QueryOutput, Variables>,
  errors: Partial<GraphQLError>[],
) {
  const handler = defineGraphQLMock<QueryOutput, Variables>(
    originalHandler.doc,
    () => {
      return {} as QueryOutput
    },
    (_output, res, ctx) => res.once(ctx.errors(errors)),
  )

  server.use(handler)
}

export function enableFeatures(features: string[]) {
  _enableFeatures(features)
}
