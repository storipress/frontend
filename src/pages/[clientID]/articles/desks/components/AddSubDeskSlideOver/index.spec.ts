import { describe, expect, it } from 'vitest'
import AddSubDeskSlideOver from './index'
import { render } from '~/test-helpers'

describe('addSubDeskSlideOver.vue', () => {
  it('props features', async () => {
    const { rerender, baseElement } = render(AddSubDeskSlideOver, {
      props: { loading: false, show: false, parent: { id: '0', name: 'Business' } },
    })
    expect(baseElement).toMatchSnapshot('hide')
    await rerender({ show: true })
    expect(baseElement).toMatchSnapshot('show')
    await rerender({ loading: true })
    expect(baseElement).toMatchSnapshot('loading')
  })
})
