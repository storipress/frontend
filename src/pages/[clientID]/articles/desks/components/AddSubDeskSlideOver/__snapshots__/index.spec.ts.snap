// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`addSubDeskSlideOver.vue > props features > hide 1`] = `
<body>
  <div>
    <!---->
  </div>
</body>
`;

exports[`addSubDeskSlideOver.vue > props features > loading 1`] = `
<body>
  <div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <div
    id="headlessui-portal-root"
  >
    
    <div
      data-headlessui-portal=""
    >
      <div
        class="fixed inset-0 z-30 pointer-events-none add-sub-desk-slide-over"
      >
        <div
          class="absolute inset-0 overflow-hidden pointer-events-none"
        >
          <!--v-if-->
          <div
            class="fixed inset-y-0 right-0 flex max-w-full pointer-events-none"
          >
            <div
              class="translate-x-full transform transition ease-out duration-500"
            >
              <form
                class="layer-2 relative z-[38] flex h-full w-screen max-w-[calc(100vw-15rem)] flex-col overflow-hidden rounded-tl-[2rem] border border-solid border-gray-100 bg-gray-100 pointer-events-auto"
              >
                <div
                  class="slide-over__header relative mx-12 border-b border-stone-200"
                >
                  
                  <button
                    class="absolute right-[-0.5rem] top-[0.375rem] z-10 inline-flex size-12 items-center justify-center rounded-full outline-none hover:bg-stone-200"
                    type="button"
                  >
                    <span
                      class="icon-cross_thin text-[1rem] text-[#4c4c4c]"
                      data-v-206d3c28=""
                    />
                  </button>
                  <div
                    class="text-pageheading py-5 text-stone-800"
                  >
                    Create New Sub-Desk in Business
                  </div>
                  
                </div>
                <div
                  class="slide-over__body mx-12 flex-shrink flex-grow overflow-auto"
                >
                  
                  <div>
                    <div
                      class="flex justify-between py-4"
                    >
                      <!-- section title wrapper -->
                      <div
                        class="w-[17rem] pt-4"
                      >
                        <!-- title -->
                        <div
                          class="text-heading pb-4 pr-2 text-stone-800"
                        >
                          
                          Subdesk settings
                          
                        </div>
                        <!-- descriptipn -->
                        <div
                          class="text-body text-stone-500"
                        >
                          
                          The subdesk name is used in your live site for section headings and navigation.
                          
                        </div>
                      </div>
                      <!-- right hand form -->
                      
                      <div
                        class="layer-1 h-fit w-[34rem] rounded-lg bg-white p-5"
                      >
                        
                        
                        <div>
                          <div>
                            <label
                              class="text-body block text-stone-800"
                              for="input-4fzzzxjylrx"
                            >
                              Subdesk name
                            </label>
                            <div
                              class="mt-1 relative"
                            >
                              <!---->
                              <div
                                class="relative"
                              >
                                <input
                                  class="text-inputs block w-full border px-3 pb-[0.312rem] pt-1.5 focus:outline-none focus:ring-1 disabled:opacity-50 rounded-md border-stone-400 focus:border-sky-600 focus:ring-sky-600"
                                  id="input-4fzzzxjylrx"
                                  name="name"
                                  type="text"
                                />
                                <!---->
                              </div>
                              
                              <!---->
                              
                            </div>
                          </div>
                        </div>
                        <div
                          class="mt-4"
                        >
                          <div>
                            <label
                              class="text-body block text-stone-800"
                              for="input-4fzzzxjylrx"
                            >
                              Subdesk description
                            </label>
                            <div
                              class="mt-1 relative"
                            >
                              <!---->
                              <div
                                class="relative"
                              >
                                <input
                                  class="text-inputs block w-full border px-3 pb-[0.312rem] pt-1.5 focus:outline-none focus:ring-1 disabled:opacity-50 rounded-md border-stone-400 focus:border-sky-600 focus:ring-sky-600"
                                  id="input-4fzzzxjylrx"
                                  name="description"
                                  type="text"
                                />
                                <!---->
                              </div>
                              
                              <!---->
                              
                            </div>
                          </div>
                        </div>
                        
                        
                      </div>
                      
                    </div>
                  </div>
                  
                </div>
                <div
                  class="slide-over__footer flex items-center border-t border-solid bg-white px-6 pb-[0.688rem] pt-[0.562rem]"
                >
                  
                  <button
                    class="text-button inline-flex h-9 items-center justify-center rounded px-4 py-2.5 focus:outline-none bg-stone-200 text-stone-500 cursor-not-allowed"
                    data-v-a52d93de=""
                    disabled=""
                    type="submit"
                  >
                    <div
                      class="flex flex-col items-center absolute"
                      data-v-a52d93de=""
                    >
                      <img
                        class="w-5 animate-spin"
                        src="data:image/svg+xml,%3csvg%20width='64'%20height='64'%20viewBox='0%200%2064%2064'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M3.776%2036.545c2.21%2013.55%2013.97%2023.89%2028.143%2023.89%2015.75%200%2028.516-12.767%2028.516-28.516%200-10.667-5.857-19.967-14.531-24.857.08-.148.687-1.176%202.33-1.566%201.293-.205%202.458.421%202.458.421a20.09%2020.09%200%200%200-.639-.465C58.48%2011.239%2064%2020.934%2064%2031.919%2064%2049.637%2049.637%2064%2031.92%2064%2017.188%2064%204.776%2054.072%201.01%2040.543c.053.17.104.331.153.482%200%200-.566-1.862%201.151-3.503.446-.36.943-.612%201.491-.756zM34.751%203.54l.214.023c-1-.106-2.017-.16-3.046-.16-3.02%200-5.93.468-8.66%201.338A27.522%2027.522%200%200%201%2034.75%203.541z'%20fill='%23047857'%20fill-rule='nonzero'/%3e%3c/svg%3e"
                      />
                      <!---->
                    </div>
                    <span
                      class="invisible flex items-center justify-center"
                      data-v-a52d93de=""
                    >
                      
                       Create new desk 
                      
                    </span>
                  </button>
                  
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</body>
`;

exports[`addSubDeskSlideOver.vue > props features > show 1`] = `
<body>
  <div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <div
    id="headlessui-portal-root"
  >
    
    <div
      data-headlessui-portal=""
    >
      <div
        class="fixed inset-0 z-30 add-sub-desk-slide-over"
      >
        <div
          class="absolute inset-0 overflow-hidden"
        >
          <div
            class="opacity-0 ease-out duration-500"
          >
            <div
              class="absolute inset-0"
            />
          </div>
          <div
            class="fixed inset-y-0 right-0 flex max-w-full"
          >
            <div
              class="translate-x-full transform transition ease-out duration-500"
            >
              <form
                class="layer-2 relative z-[38] flex h-full w-screen max-w-[calc(100vw-15rem)] flex-col overflow-hidden rounded-tl-[2rem] border border-solid border-gray-100 bg-gray-100"
              >
                <div
                  class="slide-over__header relative mx-12 border-b border-stone-200"
                >
                  
                  <button
                    class="absolute right-[-0.5rem] top-[0.375rem] z-10 inline-flex size-12 items-center justify-center rounded-full outline-none hover:bg-stone-200"
                    type="button"
                  >
                    <span
                      class="icon-cross_thin text-[1rem] text-[#4c4c4c]"
                      data-v-206d3c28=""
                    />
                  </button>
                  <div
                    class="text-pageheading py-5 text-stone-800"
                  >
                    Create New Sub-Desk in Business
                  </div>
                  
                </div>
                <div
                  class="slide-over__body mx-12 flex-shrink flex-grow overflow-auto"
                >
                  
                  <div>
                    <div
                      class="flex justify-between py-4"
                    >
                      <!-- section title wrapper -->
                      <div
                        class="w-[17rem] pt-4"
                      >
                        <!-- title -->
                        <div
                          class="text-heading pb-4 pr-2 text-stone-800"
                        >
                          
                          Subdesk settings
                          
                        </div>
                        <!-- descriptipn -->
                        <div
                          class="text-body text-stone-500"
                        >
                          
                          The subdesk name is used in your live site for section headings and navigation.
                          
                        </div>
                      </div>
                      <!-- right hand form -->
                      
                      <div
                        class="layer-1 h-fit w-[34rem] rounded-lg bg-white p-5"
                      >
                        
                        
                        <div>
                          <div>
                            <label
                              class="text-body block text-stone-800"
                              for="input-4fzzzxjylrx"
                            >
                              Subdesk name
                            </label>
                            <div
                              class="mt-1 relative"
                            >
                              <!---->
                              <div
                                class="relative"
                              >
                                <input
                                  class="text-inputs block w-full border px-3 pb-[0.312rem] pt-1.5 focus:outline-none focus:ring-1 disabled:opacity-50 rounded-md border-stone-400 focus:border-sky-600 focus:ring-sky-600"
                                  id="input-4fzzzxjylrx"
                                  name="name"
                                  type="text"
                                />
                                <!---->
                              </div>
                              
                              <!---->
                              
                            </div>
                          </div>
                        </div>
                        <div
                          class="mt-4"
                        >
                          <div>
                            <label
                              class="text-body block text-stone-800"
                              for="input-4fzzzxjylrx"
                            >
                              Subdesk description
                            </label>
                            <div
                              class="mt-1 relative"
                            >
                              <!---->
                              <div
                                class="relative"
                              >
                                <input
                                  class="text-inputs block w-full border px-3 pb-[0.312rem] pt-1.5 focus:outline-none focus:ring-1 disabled:opacity-50 rounded-md border-stone-400 focus:border-sky-600 focus:ring-sky-600"
                                  id="input-4fzzzxjylrx"
                                  name="description"
                                  type="text"
                                />
                                <!---->
                              </div>
                              
                              <!---->
                              
                            </div>
                          </div>
                        </div>
                        
                        
                      </div>
                      
                    </div>
                  </div>
                  
                </div>
                <div
                  class="slide-over__footer flex items-center border-t border-solid bg-white px-6 pb-[0.688rem] pt-[0.562rem]"
                >
                  
                  <button
                    class="text-button inline-flex h-9 items-center justify-center rounded px-4 py-2.5 text-white bg-primary hover:bg-emerald-900 transition-colors duration-200 focus:ring-emerald-500 focus:outline-none focus:ring-2 focus:ring-offset-2"
                    data-v-a52d93de=""
                    type="submit"
                  >
                    <!---->
                    <span
                      class="flex items-center justify-center"
                      data-v-a52d93de=""
                    >
                      
                       Create new desk 
                      
                    </span>
                  </button>
                  
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</body>
`;
